packages/core/src/RenderingEngine/SequentialRenderingEngine.ts

```diff
@@ -0,0 +1,377 @@
+      canvas.width = canvas.clientWidth * devicePixelRatio;
+      canvas.height = canvas.clientHeight * devicePixelRatio;
+    });
+
+    if (canvasesDrivenByVtkJs.length) {
+      this._resize(vtkDrivenViewports);
+    }
+
+    vtkDrivenViewports.forEach((vp: IStackViewport | IVolumeViewport) => {
+      const prevCamera = vp.getCamera();
+      const rotation = vp.getRotation();
+      const { flipHorizontal } = prevCamera;
+      vp.resetCameraForResize();
+
+      const displayArea = vp.getDisplayArea();
+
+      if (keepCamera) {
+        if (displayArea) {
+          if (flipHorizontal) {
+            vp.setCamera({ flipHorizontal });
+          }
+          if (rotation) {
+            vp.setViewPresentation({ rotation });
+          }
+        } else {
+          vp.setCamera(prevCamera);
+        }
+      }
+    });
+
+    if (immediate) {
+      this.render();
+    }
+  }
+
+  /**
+   * Renders all viewports.
+   */
+  protected _renderFlaggedViewports = () => {
+    this._throwIfDestroyed();
+
+    const viewports = this._getViewportsAsArray();
+    const eventDetailArray = [];
+
+    for (let i = 0; i < viewports.length; i++) {
+      const viewport = viewports[i];
+      if (this._needsRender.has(viewport.id)) {
+        const eventDetail =
+          this.renderViewportUsingCustomOrVtkPipeline(viewport);
+        eventDetailArray.push(eventDetail);
+        viewport.setRendered();
+
+        this._needsRender.delete(viewport.id);
+
+        if (this._needsRender.size === 0) {
+          break;
+        }
+      }
+    }
+
+    this._animationFrameSet = false;
+    this._animationFrameHandle = null;
+
+    eventDetailArray.forEach((eventDetail) => {
+      if (!eventDetail?.element) {
+        return;
+      }
+      triggerEvent(eventDetail.element, Events.IMAGE_RENDERED, eventDetail);
+    });
+  };
+
+  /**
+   * Renders the given viewport
+   * using its proffered method.
+   *
+   * @param viewport - The viewport to render
+   */
+  protected renderViewportUsingCustomOrVtkPipeline(
+    viewport: IViewport
+  ): EventTypes.ImageRenderedEventDetail {
+    let eventDetail: EventTypes.ImageRenderedEventDetail;
+
+    if (
+      viewport.sWidth < VIEWPORT_MIN_SIZE ||
+      viewport.sHeight < VIEWPORT_MIN_SIZE
+    ) {
+      console.warn('Viewport is too small', viewport.sWidth, viewport.sHeight);
+      return;
+    }
+    if (viewportTypeUsesCustomRenderingPipeline(viewport.type) === true) {
+      eventDetail =
+        viewport.customRenderViewportToCanvas() as EventTypes.ImageRenderedEventDetail;
+    } else {
+      if (this.useCPURendering) {
+        throw new Error(
+          'GPU not available, and using a viewport with no custom render pipeline.'
+        );
+      }
+
+      const { offscreenMultiRenderWindow } = this;
+      const renderWindow = offscreenMultiRenderWindow.getRenderWindow();
+
+      this._resizeOffScreenCanvasForSingleViewport(viewport.canvas);
```

Comment:
Can we just keep it the same size as the largest viewport all the time?

And only resize it when any of these are resized?